astropy/coordinates/representation.py

```diff
@@ -219,6 +219,21 @@ def to_cartesian(self):
         # Note: the above docstring gets overridden for differentials.
         raise NotImplementedError()
 
+    def to_cartesian_with_differentials(self):
```

Comment:
Am still not too keen to add this to `representation`, as I don't like the duplication of `to_cartesian` and find the name rather long (but don't have a better suggestion).

How about using it for `.cartesian` on a frame, though? For that case, I feel there is no doubt one wants to transform the derivatives as well, and we know that we only have first-order to worry about. I guess that means adapting `represent_as`. We can always decide to move it down to `representation` later.